Obsidian/Commands/MainCommandModule.cs

```diff
@@ -493,4 +493,57 @@ private ChatMessage SendCommandUsage(string commandUsage)
         commands.AddExtra(usage);
         return commands;
     }
+
+    [Command("give")]
+    [CommandInfo("Give an item for a specific player", "/give <player> <block> [<amount>]")]
+    public async Task GiveAsync(CommandContext ctx, string player, string block) => await GiveAsync(ctx, player, block, 1);
+
+    [CommandOverload]
+    public async Task GiveAsync(CommandContext ctx, string player, string block, short amount)
+    {
+        try
+        {
+            IReadOnlyList<Player> players = new List<Player>();
+            if(player.EqualsIgnoreCase("@a")) players = ctx.Server.Players.Select(p=>p as Player).ToList();
+            if(!player.StartsWith("@") /*&& player.IsValidUsername()*/) players = new List<Player> { ctx.Server.GetPlayer(player, StringComparison.OrdinalIgnoreCase) as Player };
+            
+            var prefix = block.Contains(":") ? block.Split(":")[0] : "minecraft";
+
+            if (block.StartsWith("minecraft:"))
+            {
+                block = block[prefix.Length..];
+            }
+            else if (block.Contains(":"))
+            {
+                await ctx.Sender.SendMessageAsync($"{ChatColor.Red}Only vanilla items are supported.");
+                return;
+            }
+
+            if (!Enum.TryParse<Material>(block.Replace("_", ""), true, out var material))
+            {
+                if (int.TryParse(block, out var materialId))
+                {
+                    material = (Material)materialId;
+                }
+                else
+                {
+                    await ctx.Sender.SendMessageAsync($"{ChatColor.Red}Specified item {block} not found.");
+                    return;
+                }
+            }
```

Comment:
This feels hacky, this should be addressed in the future. I don't see a proper block ID system here.

Reply:
I created the `/give` command when I started working on signs and then I thought that it might be useful for someone. I agree, its a hacky code.